### DISCUSSION TITLE
Expose useful functions (openAccountModal, etc) via hook(s)

### DISCUSSION THREAD

**Discussion by johnrjj** [rainbow-me/rainbowkit#502] [Ideas]
Would prefer to use hooks vs ConnectButton.Custom HOC

**Comment by johnrjj** [comment]
Ah this has been proposed already -- https://github.com/rainbow-me/rainbowkit/discussions/432

Not sure how to close discussions.

**Comment by markdalgleish** [comment]
This is out now in [RainbowKit v0.4.3.](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.4.3)

You can check out the docs here: https://www.rainbowkit.com/docs/modal-hooks

**Reply by johnrjj** [reply]
Thanks, Mark!! Will be consuming these hooks this week! :)

**Reply by chcharcharlie** [reply]
Hey Mark @markdalgleish , I was trying these hooks out however weren't able to trigger it. Did some console loggings and found that `openConnectModal` and `openAccountModal` has always been undefined even though the <ConnectButton> works fine.

Wondering if there are some known common causes for this issue? I'm guessing one of the following but can't be sure:
1. Do the hooks need to be used inside the same component where the <ConnectButton> is rendered?
2. Could there be any issue if I used SIWE authentication and had a custom authentication adaptor?

Thank you!

**Comment by JinNguyen96** [comment]
I keeps getting undefined for openConnectWallet from hook after 1 failed connection. Any idea about this?
I tried to connect via Abstract global wallet, then I got an RPC error, later then, I can no longer trigger the openConnectWallet anymore
